From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qdio: prevent double qdio shutdown in case of I/O errors.
References: bnc#445100

Symptom:     I/O errors reported using a zfcp disk.
Problem:     In case of I/O errors on a qdio subchannel qdio_shutdown may be 
             called twice by the qdio driver and by zfcp. 
Solution:    Remove the superfluous shutdown from qdio and let the upper layer driver
             handle the error condition.

Acked-by: John Jolly <jjolly@suse.de>

---
 drivers/s390/cio/qdio_main.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-sles11/drivers/s390/cio/qdio_main.c
===================================================================
--- linux-sles11.orig/drivers/s390/cio/qdio_main.c
+++ linux-sles11/drivers/s390/cio/qdio_main.c
@@ -1080,7 +1080,6 @@ void qdio_int_handler(struct ccw_device 
 		case -EIO:
 			sprintf(dbf_text, "ierr%4x", irq_ptr->schid.sch_no);
 			QDIO_DBF_TEXT2(1, setup, dbf_text);
-			qdio_int_error(cdev);
 			return;
 		case -ETIMEDOUT:
 			sprintf(dbf_text, "qtoh%4x", irq_ptr->schid.sch_no);
